src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-cluster.cpp

```diff
@@ -0,0 +1,372 @@
+        {
+            Attributes::PacketMulticastRxCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiPacketMulticastRxCount, count, encoder);
+        }
+        break;
+    }
+    case PacketMulticastTxCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kPacketCounts) && mLogic.GetEnabledAttributes().enablePacketMulticastTxCount)
+        {
+            Attributes::PacketMulticastTxCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiPacketMulticastTxCount, count, encoder);
+        }
+        break;
+    }
+    case PacketUnicastRxCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kPacketCounts) && mLogic.GetEnabledAttributes().enablePacketUnicastRxCount)
+        {
+            Attributes::PacketUnicastRxCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiPacketUnicastRxCount, count, encoder);
+        }
+        break;
+    }
+    case PacketUnicastTxCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kPacketCounts) && mLogic.GetEnabledAttributes().enablePacketUnicastTxCount)
+        {
+            Attributes::PacketUnicastTxCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiPacketUnicastTxCount, count, encoder);
+        }
+        break;
+    }
+    case CurrentMaxRate::Id: {
+        if (mLogic.GetEnabledAttributes().enableCurrentMaxRate)
+        {
+            Attributes::CurrentMaxRate::TypeInfo::Type rate;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiCurrentMaxRate, rate, encoder);
+        }
+        break;
+    }
+    case OverrunCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kErrorCounts) && mLogic.GetEnabledAttributes().enableOverrunCount)
+        {
+            Attributes::OverrunCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiOverrunCount, count, encoder);
+        }
+        break;
+    }
+    default:
+        break;
+    }
+    return Protocols::InteractionModel::Status::UnsupportedAttribute;
+}
+
+CHIP_ERROR WiFiDiagnosticsServer::Attributes(const ConcreteClusterPath & path,
+                                             ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
+{
+    // mandatory attributes
+    ReturnErrorOnFailure(builder.AppendElements({
+        Bssid::kMetadataEntry,
+        SecurityType::kMetadataEntry,
+        WiFiVersion::kMetadataEntry,
+        ChannelNumber::kMetadataEntry,
+        Rssi::kMetadataEntry,
+    }));
+
+    if (mLogic.GetFeatureFlags().Has(Feature::kErrorCounts))
+    {
+        ReturnErrorOnFailure(builder.AppendElements({
+            BeaconLostCount::kMetadataEntry,
+            OverrunCount::kMetadataEntry,
+        }));
+    }
+    if (mLogic.GetFeatureFlags().Has(Feature::kPacketCounts))
```

Comment:
OK, so here we claim to support BeaconRxCount just based on the feature, which is good.  Then why do we have `GetEnabledAttributes().enableBeaconRxCount` as a thing at all?

Reply:
Removed it.